### ISSUE TITLE
[Bug]: Fast Pass accessibility tool (Accessibility insights for web) marks Detailslist as not compliant

### ISSUE THREAD

**Issue by irinatarn** [microsoft/fluentui#28706]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (16) x64 Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
    Memory: 29.61 GB / 63.95 GB
  Browsers:
    Edge: Spartan (44.22621.1992.0), Chromium (115.0.1901.188)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist

## Actual Behavior
Please go to the link above and execute Accessibility Insights for Web (a tool used to detect accessibility issues)
![image](https://github.com/microsoft/fluentui/assets/28061748/0d3c63f1-cc7f-4c96-be93-25d003830fce)

## Expected Behavior
Component shouldn't fail with aria-required-children.

### Requested priority

Blocking

### Products/sites affected

dfp.microsoft.com (and any site using fluent details list)

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by sopranopillow** [comment]
Looks like this is a false positive, axe core should ignore items hidden from screen readers as mentioned here https://github.com/dequelabs/axe-core/issues/3850 (Note this is not solved even though it's closed). I've highlighted the problems in this picture:
![image](https://github.com/microsoft/fluentui/assets/5953191/c69a3e87-d39e-4ea6-86b0-7a4d42adc10a)

These have `aria-hidden`, therefore it does not show up for screen reader users, I'll let @smhigley confirm this, but there's nothing we can do on our side.

**Comment by smhigley** [comment]
Yup! That was a false positive in axe-core that has since been fixed. It just needs to be picked up by accessibility insights, and then the error will go away. For those internal to Microsoft, we have it documented on our [known false positives wiki](https://dev.azure.com/microsoftdesign/fluent-ui/_wiki/wikis/fluent-ui.wiki/332/Fluent-v8-known-issues)
